### DISCUSSION TITLE
Added: LDM Notes for Oct. 23rd and Oct 28th, 2019

### DISCUSSION THREAD

**Discussion by agocke** [dotnet/csharplang#2921] [General]
https://github.com/dotnet/csharplang/blob/master/meetings/2019/LDM-2019-10-23.md

1. New primitive types (`nint` and `nuint`)

https://github.com/dotnet/csharplang/blob/master/meetings/2019/LDM-2019-10-28.md

1. Discard parameters in lambdas and other methods
2. Enhancing the common type algorithm

**Comment by tannergooding** [comment]
I think the one thing missing from `nint`/`nuint` is the discussion LDM had around shifts and masking. That is, current shifts for int/uint mask the shift count to 31 and long/ulong to 63. We can't mask with a constant for `nint`/`nuint` and would either need to do `(sizeof(nint) * 8) - 1` or have the framework expose some new "well-known" member.

I picked up the work item to bring forth an API proposal on corefx.
